fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3854,6 +3928,10 @@ public Builder setMetaDataProvider(@Nullable RecordMetaDataProvider metaDataProv
             return this;
         }
 
+        public void setMetaDataPreloadFuture(@Nonnull CompletableFuture<Void> metaDataPreloadFuture) {
+            this.metaDataPreloadFuture = metaDataPreloadFuture;
+        }
```

Comment:
It looks like the meta-data preload future always gets overridden in `uncheckedOpenAsync` and in `createOrOpenAsync` (but not in `build()`, I suppose). I don't think that the values chosen there are necessarily wrong (and simplifies the question of "what to do if the meta-data and the preload future don't match"), but it's possible we don't want the user to be able to set this (yet?).

Reply:
sure, dropped it.